front/src/common/Map/Search/MapSearchOperationalPoint.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState, useEffect } from 'react';
+import { getMap } from 'reducers/map/selectors';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { onResultSearchClick } from 'common/Map/utils';
+import { useDebounce } from 'utils/helpers';
+import { useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import cx from 'classnames';
+
+import InputSNCF from 'common/BootstrapSNCF/InputSNCF';
+
+import type { SearchResultItemOperationalPoint } from 'common/api/osrdEditoastApi';
+import { useAppDispatch } from 'store';
+import type { Viewport } from 'reducers/map';
+import { useInfraID } from 'common/osrdContext';
+import CheckboxRadioSNCF from 'common/BootstrapSNCF/CheckboxRadioSNCF';
+
+const mainOperationalPointsCHCodes = ['', '00', 'BV'];
+
+type MapSearchOperationalPointProps = {
+  updateExtViewport: (viewport: Partial<Viewport>) => void;
+  closeMapSearchPopUp: () => void;
+};
+
+const MapSearchOperationalPoint = ({
+  updateExtViewport,
+  closeMapSearchPopUp,
+}: MapSearchOperationalPointProps) => {
+  const map = useSelector(getMap);
+  const [searchResults, setSearchResults] = useState<SearchResultItemOperationalPoint[]>([]);
+  const [orderedResults, setOrderedResults] = useState<SearchResultItemOperationalPoint[]>([]);
+  const [chCodeFilter, setChCodeFilter] = useState('');
+  const [mainOperationalPointsOnly, setMainOperationalPointsOnly] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
+  const infraID = useInfraID();
+
+  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
+  const dispatch = useAppDispatch();
+
+  const { t } = useTranslation(['map-search']);
+
+  const debouncedSearchTerm = useDebounce(searchTerm, 500);
+
+  const sortAndFilterSearchResults = () =>
+    [...searchResults]
+      .map((result) => ({
+        ...result,
+        // remove CH Code information when it's a main operational point (=== "BV" or "00") to ensure it'll be on top of search results
+        ch: mainOperationalPointsCHCodes.includes(result.ch) ? '' : result.ch,
+      }))
+      // Begin to filter with main operational points (CH code = ''), if not checked, filter on chCode input field
+      .filter((result) => {
+        if (mainOperationalPointsOnly) return result.ch === '';
+        return chCodeFilter !== ''
+          ? chCodeFilter.trim().toLowerCase() === result.ch.toLocaleLowerCase()
+          : true;
+      })
+      .sort((a, b) => a.name.localeCompare(b.name) || a.ch.localeCompare(b.ch));
```

Comment:
We could avoid the `orderedResults` useState and just put this in a useMemo with the same dependency array as the useEffect line 97.

Reply:
Yes, you're right! Done.